library/public/options.h

```diff
@@ -0,0 +1,107 @@
+/**
+ * @class   options
+ * @brief   Class used to control the different options
+ *
+ * A class to control the different options of f3d.
+ * See the README_libf3d.md for the full listing of options
+ */
+
+#ifndef f3d_options_h
+#define f3d_options_h
+
+#include "export.h"
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace f3d
+{
+class F3D_EXPORT options
+{
+public:
+  //@{
+  /**
+   * Default/Copy/move constructors/operators
+   */
+  options();
+  ~options();
+  options(const options& opt);
+  options& operator=(const options& opt) noexcept;
+  options(options&& other) noexcept;
+  options& operator=(options&& other) noexcept;
+  //@}
+
+  class exception : public std::runtime_error
+  {
+  public:
+    exception(const std::string& what = "")
+      : std::runtime_error(what)
+    {
+    }
+  };
+
+  //@{
+  /**
+   * Setters for all supported types
+   */
+  options& set(const std::string& name, bool value);
```

Comment:
Explicitly listing the options may be sensible. It would allow to maintain associated documentation & deprecation scheme as well as writing python code where such option would auto-complete.

Reply:
Say differently, available options available for each type of objects (model, ui, scene, ...) should probably be query-able. The associated description and type retrievable as well.

Reply:
This is exactly what we do not want. By having a string based API we allow ouselves to add/deprecate options with minimum effort, from both lib dev and lib users.

However, having a way to query available options and generate automatically python code could be considered.

Reply:
Having a specialized domain for specifying options would reduce chance of error and would help integrate and use the library in application.

Using any of these would streamline usage, reduce confusion and be less error prone:
* enum classes
* constant to reference the option key
* strongly type object allowing to easily know if an option is a boolean, integer, double, ...

Reply:
Wdyt @Meakk ?

Reply:
We already had this discussion, it's all about strong typing versus flexibility. With the current implementation, the api is much simpler, but yes, the user has to know about the options and will see errors at runtime only. We can discuss it offline again @mwestphal